proxy/hdrs/HdrHeap.h

```diff
@@ -150,20 +145,22 @@ class HdrStrHeap : public RefCountObj
   char *m_free_start;
   uint32_t m_free_size;
 
-  bool
-  contains(const char *str) const
-  {
-    return (str >= ((const char *)this + STR_HEAP_HDR_SIZE) && str < ((const char *)this + m_heap_size));
-  }
+  bool contains(const char *str) const;
 };
 
+inline bool
+HdrStrHeap::contains(const char *str) const
+{
+  return reinterpret_cast<char const *>(this + 1) <= str && str < reinterpret_cast<char const *>(this) + m_heap_size;
```

Comment:
Why do you replace STR_HEAP_HDR_SIZE with 1 in the expression?  Was STR_HEAP_HDR_SIZE incorrect in the original expression?

Reply:
The key is when exactly the cast is done - in the original code `this` is converted to a `char*` and then the struct size is added. In this case, `this` is incremented by 1 first and then converted to `char*`. These, by language definition, are equivalent. However, the latter lets me get rid of a `#define` and therefore (IMHO) is overall cleaner.

Reply:
Ok.  Missed the parenthesis changes.  Makes sense.